### PR TITLE
Fix problematic type comments

### PR DESCRIPTION
**Pull request by zetaraku**
A previous commit introduced some comments that didn't reflect the real implementation which may lead to unsafe assumptions of default behaviors:

https://github.com/fastify/fastify-cookie/commit/04ea01fbede66e409889b26c7e59ffd3468bab92#diff-0647ab7efe94459b8666eed6d1660236e363af58bb3fa7eda85041a316c97bb2R108-R126

This PR fixes that.

#### Checklist

- [ ] run `npm run test` and `npm run benchmark`
- [ ] tests and/or benchmarks are included
- [x] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Comment by mcollina** [comment]
If the comments are incorrect, could you fix them instead of removing?

File: types/plugin.d.ts [hunk 1]
```diff
@@ -129,11 +129,12 @@ declare namespace fastifyCookie {
     sameSite?: 'lax' | 'none' | 'strict' | boolean;
     /** One of the `Priority` string attributes (`low`, `medium` or `high`) specifying a retention priority for HTTP cookies that will be respected by user agents during cookie eviction. */
     priority?: 'low' | 'medium' | 'high';
-    /** The `boolean` value of the `Secure` attribute. Set this option to false when communicating over an unencrypted (HTTP) connection. Value can be set to `auto`; in this case the `Secure` attribute will be set to false for HTTP request, in case of HTTPS it will be set to true. */
+    /** Add the `Secure` attribute. Defaults to `false`. */
```
Comment (Uzlopak) [inline types/plugin.d.ts:132]: How does this improve it?

Comment (zetaraku) [reply types/plugin.d.ts:132]: `"auto"` is only available in `CookieSerializeOptions`, so it's moved there. 👀

**Review by mcollina — APPROVED** [review]
lgtm

**Review by gurgunday — APPROVED** [review]

**Pull request merged**